File: frontend/src/types/analysis.ts
import type { StockQuote, TechnicalIndicators } from "./stock";
import type { Fundamentals, FundamentalScore } from "./fundamentals";
import type { TradeSignal } from "./signal";

export interface FullAnalysisResponse {
  quote: StockQuote;
  indicators: TechnicalIndicators;
  fundamentals: Fundamentals;
  score: FundamentalScore;
  signal: TradeSignal;
}

File: backend/app/schemas/analysis.py
from typing import Optional
from app.schemas.base import BaseSchema
from app.schemas.stock import StockQuote, TechnicalIndicators
from app.schemas.fundamentals import Fundamentals, FundamentalScore
from app.schemas.signal import TradeSignal


class FullAnalysisResponse(BaseSchema):
    quote: StockQuote
    indicators: TechnicalIndicators
    fundamentals: Fundamentals
    score: FundamentalScore
    signal: TradeSignal
